cmd/chainsimulator/main.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/core/closing"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-chain-logger-go/file"
+	"github.com/multiversx/mx-chain-simulator-go/config"
+	"github.com/multiversx/mx-chain-simulator-go/pkg/facade"
+	endpoints "github.com/multiversx/mx-chain-simulator-go/pkg/proxy/api"
+	"github.com/multiversx/mx-chain-simulator-go/pkg/proxy/configs"
+	"github.com/multiversx/mx-chain-simulator-go/pkg/proxy/creator"
+	"github.com/urfave/cli"
+)
+
+var (
+	log          = logger.GetOrCreate("chainsimulator")
+	helpTemplate = `NAME:
+   {{.Name}} - {{.Usage}}
+USAGE:
+   {{.HelpName}} {{if .VisibleFlags}}[global options]{{end}}
+   {{if len .Authors}}
+AUTHOR:
+   {{range .Authors}}{{ . }}{{end}}
+   {{end}}{{if .Commands}}
+GLOBAL OPTIONS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+VERSION:
+   {{.Version}}
+   {{end}}
+`
+)
+
+func main() {
+	app := cli.NewApp()
+	cli.AppHelpTemplate = helpTemplate
+	app.Name = "Chain Simulator"
+	app.Usage = ""
+	app.Flags = []cli.Flag{
+		configurationFile,
+		logLevel,
+		logSaveFile,
+		disableAnsiColor,
+	}
+
+	app.Authors = []cli.Author{
+		{
+			Name:  "The MultiversX Team",
+			Email: "contact@multiversx.com",
+		},
+	}
+
+	app.Action = startChainSimulator
+
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+	}
+}
+
+// TODO implement a mechanism that will fetch the config base on the go.mod --> for node and proxy
+const (
+	pathToNodeConfig  = "../../../mx-chain-go/cmd/node/config"
+	pathToProxyConfig = "../../../mx-chain-proxy-go/cmd/proxy/config"
+)
+
+func startChainSimulator(ctx *cli.Context) error {
+	cfg, err := loadMainConfig(ctx.GlobalString(configurationFile.Name))
+	if err != nil {
+		return fmt.Errorf("%w while loading the config file", err)
+	}
+	fileLogging, err := initializeLogger(ctx, cfg)
+	if err != nil {
+		return fmt.Errorf("%w while initializing the logger", err)
+	}
+
+	startTime := time.Now().Unix()
+	roundDurationInMillis := uint64(6000)
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	apiConfigurator := api.NewFreePortAPIConfigurator("localhost")
```

Comment:
suggestion to parametrize this in a next PR

Reply:
it's done in the next PR